api/soap/mc_issue_api.php

```diff
@@ -90,7 +90,243 @@ function mc_issue_get( $p_username, $p_password, $p_issue_id ) {
 }
 
 /**
-* Get history details about an issue.
+ * Get history details for an issue that is visible to specified user.
+ * This is used by REST APIs and assumes that user access to issue is
+ * already done.
+ *
+ * @param integer $p_issue_id The issue id.
+ * @param integer $p_user_id The user id.
+ * @param string $p_lang The user language.
+ * @return array history entries or empty if user has no access to history.
+ */
+function mci_issue_get_history( $p_issue_id, $p_user_id, $p_lang ) {
+	$t_project_id = bug_get_field( $p_issue_id, 'project_id' );
+
+	$t_view_history_threshold = config_get( 'view_history_threshold', null, null, $t_project_id );
+	if( !access_has_bug_level( $t_view_history_threshold, $p_issue_id, $p_user_id ) ) {
+		return array();
+	}
+
+	$t_history_rows = history_get_raw_events_array( $p_issue_id, $p_user_id );
+
+	$t_history = array();
+
+	foreach( $t_history_rows as $t_history_row ) {
+		$t_type = (int)$t_history_row['type'];
+
+		$t_skip = false;
+
+		switch( $t_type ) {
+			case BUG_ADD_SPONSORSHIP:         # Deprecated, not exposed in REST API
+			case BUG_UPDATE_SPONSORSHIP:      # Deprecated, not exposed in REST API
+			case BUG_DELETE_SPONSORSHIP:      # Deprecated, not exposed in REST API
+			case BUG_REVISION_DROPPED:        # Not Supported
+			case BUGNOTE_REVISION_DROPPED:    # Not Supported
+				$t_skip = true;
```

Comment:
not needed

Reply:
For some reason `continue;` doesn't work as expected, to make it clear I replaced `continue` with `break` and kept the setting of skip.